### ISSUE TITLE
Slices don't work when converting to ONNX and using ONNX.js

### ISSUE THREAD

**Issue by elliotwaite** [pytorch/pytorch#32956]
## 🐛 Bug

If I convert a model that uses a slice (such as `x = x[:, 0]`) into an ONNX model, and then use that ONNX model with ONNX.js, I get an error. I am not sure if this is an issue with the PyTorch-to-ONNX conversion or with the ONNX.js library.

## To Reproduce

Generate an ONNX model with the script below:

```
import torch
import torch.nn as nn


class Model(nn.Module):
  def forward(self, x):
    x = x[:, 0]
    return x


def main():
  model = Model()
  dummy_input = torch.zeros(1, 2)
  torch.onnx.export(model, dummy_input, 'model.onnx', verbose=True)


if __name__ == '__main__':
  main()
```

This will generate this graph:

```
graph(%0 : Float(1, 2)):
  %1 : Tensor = onnx::Constant[value={0}]()
  %2 : Float(1) = onnx::Gather[axis=1](%0, %1)
  return (%2)
```

This will also generate the `model.onnx` file. We can then run the below HTML code in the same directory as the `model.onnx` file to try to use the model with ONNX.js:

```
<html>
  <body>
    <script src="https://cdn.jsdelivr.net/npm/onnxjs/dist/onnx.min.js"></script>
    <script>
      async function test() {
        const sess = new onnx.InferenceSession()
        await sess.loadModel('./model.onnx')
        const input = new onnx.Tensor(new Float32Array(2), 'float32', [1, 2])
        const outputMap = await sess.run([input])
        const outputTensor = outputMap.values().next().value
        console.log(`Output tensor: ${outputTensor.data}`)
      }
      test();
    </script>
  </body>
</html>
```

This will produce the following console log:

```
Uncaught (in promise) Error: Failed to compile shader: ERROR: 0:87: '' : array size must be greater than zero

    at t.compileShader (webgl-context.ts:174)
    at t.compile (program-manager.ts:97)
    at program-manager.ts:67
    at t.event (instrument.ts:294)
    at t.build (program-manager.ts:64)
    at t.run (inference-handler.ts:26)
    at e.run (gather.ts:10)
    at t.<anonymous> (execution-plan.ts:104)
    at backend.ts:102
    at Object.next (backend.ts:102)
```

To get the expected behavior, we can swap out using a slice (`x[:, 0]`) for using `narrow` (`torch.narrow(x, dim=1, start=1, length=1)`) as in the script below:

```
import torch
import torch.nn as nn


class Model(nn.Module):
  def forward(self, x):
    x = torch.narrow(x, dim=1, start=1, length=1)
    return x


def main():
  model = Model()
  dummy_input = torch.zeros(1, 2)
  torch.onnx.export(model, dummy_input, 'model.onnx', verbose=True)


if __name__ == '__main__':
  main()
```

This will generate this graph:
```
graph(%0 : Float(1, 2)):
  %1 : Float(1, 1) = onnx::Slice[axes=[1], ends=[2], starts=[1]](%0)
  return (%1)
```

And using the generated ONNX model with the same HTML code as above will work, with a console log output of:
```
Output tensor: 0
```

## Environment
PyTorch version: 1.5.0.dev20200203
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.6
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.5.0.dev20200203
[pip] torchvision==0.6.0a0+a5525e9
[conda] blas 1.0 mkl
[conda] mkl 2019.4 233
[conda] mkl-service 2.3.0 py37hfbe908c_0
[conda] mkl_fft 1.0.15 py37h5e564d8_0
[conda] mkl_random 1.1.0 py37ha771720_0
[conda] pytorch 1.5.0.dev20200203 py3.7_0 pytorch-nightly
[conda] torchvision 0.6.0.dev20200203 py37_cpu pytorch-nightly

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by spandantiwari** [comment]
@elliotwaite - do you see anything in the ONNX graph that seems incorrect/invalid?

**Comment by elliotwaite** [comment]
It looks valid to me, but I'm not very familiar with verifying ONNX graphs. I've opened the issue with ONNX.js as well to see if they have any insights: https://github.com/microsoft/onnxjs/issues/179

**Comment by garymm** [comment]
FYI ONNX.js became ONNX Runtime for Web so if you still have this issue you should try it on ONNX Runtime and report problems there.
Closing since it doesn't seem like `torch.onnx` is buggy here.

**Comment by elliotwaite** [comment]
Thanks for the update. I just confirmed that this issue has been resolved for ONNX Runtime Web.
